src/runtime/float.cpp

```diff
@@ -868,27 +868,15 @@ template <ExceptionStyle S> static BoxedFloat* _floatNew(Box* a) noexcept(S == C
 
         return res;
     } else {
-        static BoxedString* float_str = internStringImmortal("__float__");
```

Comment:
and if this change was not only to remove code duplication: could you maybe add a small test case which now works but did not before